shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -74,23 +77,24 @@
         private final UUID id;
         private long lastEventNumber = -1L;
         private int eventCount;
-        private long requestId;
+        private final long requestId;
     }
 
     @Override
     protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws Exception {
         log.trace("Encoding message to send over the wire {}", msg);
         if (msg instanceof Append) {
             Append append = (Append) msg;
-            Session session = setupSegments.get(append.segment);
+                Session session = setupSegments.get(new SimpleImmutableEntry<>(append.segment, append.getWriterId()));
```

Comment:
Is there any change to this class that makes it backward incompatible?

Reply:
No, these changes do not make it backward incompatible.